tools/cloud-function-automation/cloud_function_automation.py

```diff
@@ -0,0 +1,278 @@
+    :param project_num: Google Cloud project number mentioned in variables.py.
+    :param token: Google Cloud auth token.
+    :return: True or False.
+    """
+    url = f"https://serviceusage.googleapis.com/v1/projects/{project_num}/services:batchEnable"
+
+    payload = json.dumps({
+        "serviceIds": [
+            "pubsub.googleapis.com",
+            "cloudfunctions.googleapis.com",
+            "logging.googleapis.com"
+        ]
+    })
+    headers = {
+        'Authorization': token,
+        'Content-Type': 'application/json'
+    }
+    try:
+        response = requests.request("POST", url, headers=headers, data=payload)
+    except requests.exceptions.Timeout as exc:
+        raise SystemExit(f"Timeout Error while calling enable service API - {exc}")
+    except requests.exceptions.TooManyRedirects as exc:
+        raise SystemExit(f"TooManyRedirects Error while calling enable service API - {exc}")
+    except requests.exceptions.RequestException as exc:
+        raise SystemExit(f"RequestException while calling enable service API - {exc}")
+
+    if response.status_code == 200:
+        services_enabled = True
+    else:
+        services_enabled = False
+
+    return services_enabled
+
+
+def create_pubsub_topic(project_id, pubsub_topic_name):
+    """
+    This function will be used to create pubsub topic if not already present.
+    :param project_id: Google Cloud project id mentioned in variables.py.
+    :param pubsub_topic_name: Google Cloud Pub/Sub topic name mentioned in variables.py.
+    :return: True or False.
+    """
+    try:
+        publisher = pubsub_v1.PublisherClient()
+    except Exception as exc:
+        raise SystemExit(exc)
+    full_topic_name = f'projects/{project_id}/topics/{pubsub_topic_name}'
+    try:
+        publisher.create_topic(name=full_topic_name)
+        print(f"PubSub Topic {full_topic_name} has been created")
+        created_pubsub_topic = True
+    except AlreadyExists:
+        print(f"{full_topic_name} already exists.")
+        created_pubsub_topic = False
+    return created_pubsub_topic
+
+
+def create_sink(sink_name, full_topic_name, log_sink_filter):
+    """
+    This function will be used to create Log Sink from Log Router.
+    :param sink_name: Log Sink name mentioned in variables.py.
+    :param full_topic_name: Pub/Sub topic associated with log sink.
+    :param log_sink_filter: log sink filer mentioned in variables.py.
+    :return: True or False.
+    """
+    try:
+        logging_client = logging.Client()
+    except Exception as exc:
+        raise SystemExit(exc)
+    destination = f"pubsub.googleapis.com/{full_topic_name}"
+    sink = logging_client.sink(sink_name, filter_=log_sink_filter, destination=destination)
+    if sink.exists():
+        print(f"Log Router Sink {sink.name} already exists.")
+        created_sink = False
+    else:
+        try:
+            sink.create()
+        except Exception as exc:
```

Comment:
Same here (comments was: What kind of exception can happen here? Handling a generic exception and then exiting immediately does not do much. Please handle the exceptions that you would know how to handle, or at least, that you could enrich with some more meaningful error message.)